miniapps/shifted_methods/dist_solver.hpp

```diff
@@ -0,0 +1,322 @@
+// Copyright (c) 2010-2020, Lawrence Livermore National Security, LLC. Produced
+// at the Lawrence Livermore National Laboratory. All Rights reserved. See files
+// LICENSE and NOTICE for details. LLNL-CODE-806117.
+//
+// This file is part of the MFEM library. For more information and source code
+// availability visit https://mfem.org.
+//
+// MFEM is free software; you can redistribute it and/or modify it under the
+// terms of the BSD-3 license. We welcome feedback and contributions, see file
+// CONTRIBUTING.md for details.
+
+#ifndef MFEM_DIST_FUNCTION_HPP
+#define MFEM_DIST_FUNCTION_HPP
+
+#include "mfem.hpp"
+
+namespace mfem
+{
+
+class DistanceSolver
```

Comment:
A comment block would be helpful here.  I can guess what `ComputeScalarDistance` might do but I really don't know what `ComputeVectorDistance` does.  Some overview of the role this class plays in the algorithm would be appreciated.  Also, if there's a paper that describes the algorithm in detail it would be good to have a reference to that paper here.

Reply:
I agree with Mark. I think we need better documentation for all the Classes in this file.